### PR TITLE
[MM-41576] Revamp database schema version

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
- Add two new methods to `sqlStore.Store`:
  - `GetDBSchemaVersion` To get last applied migration number
  - `GetAppliedMigrations` To get all applied migrations (this will also going to be used in the admin console)
- Add new `db version` subcommand to get schema version. Also add `--all` flag to get all applied migrations`. 

We may want to skip starting migrations for `db version` sub command. And also we should remove `upgrade.go` entirely.  Let me know what you think. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-41576

#### Release Note

```release-note
Adds new db version subcommand to query database schema version
```

### CODE REVIEW

**Comment by agnivade** [comment]
> And also we should remove upgrade.go entirely.

Heavy +1 to that.

Also, will you be working on the webapp changes? I'd like to see how it looks. :)

**Comment by isacikgoz** [comment]
@agnivade it's on the way 😛

**Comment by agnivade** [comment]
I'll review this along with the webapp PR when it comes out.

File: app/config.go [hunk 1]
```diff
@@ -357,6 +357,11 @@ func (a *App) ClientConfigWithComputed() map[string]string {
 	if installationDate, err := a.ch.srv.getSystemInstallDate(); err == nil {
 		respCfg["InstallationDate"] = strconv.FormatInt(installationDate, 10)
 	}
+	if ver, err := a.ch.srv.Store.GetDBSchemaVersion(); err != nil {
+		mlog.Error("Could not get the schema version", mlog.Err(err))
+	} else {
+		respCfg["Version"] = strconv.Itoa(ver)
```
Comment (agnivade) [inline app/config.go:363]: Is this the value that shows up in the modal? I am wondering how was the webapp showing the previous value? I don't see any code edited here. :thinking:

File: store/sqlstore/store_test.go [hunk 2]
```diff
@@ -784,3 +784,41 @@ func TestMySQLReadTimeout(t *testing.T) {
...
+
+	for _, driver := range testDrivers {
+		t.Run("Should return latest version number of applied migrations for "+driver, func(t *testing.T) {
+			t.Parallel()
+			settings := makeSqlSettings(driver)
+			store := New(*settings, nil)
+
+			version, err := store.GetDBSchemaVersion()
+			require.NoError(t, err)
+			require.NotZero(t, version)
```
Comment (agnivade) [inline store/sqlstore/store_test.go:802]: Wondering if we can read the migrations dir, find out the latest numbered sql file and match that with the value here? Or will that be too cumbersome? It seems like a `NotZero` test is too weak, but I don't want to have a super hacky check either. LMK.

File: store/sqlstore/store_test.go [hunk 3]
```diff
@@ -784,3 +784,41 @@ func TestMySQLReadTimeout(t *testing.T) {
...
+
+	for _, driver := range testDrivers {
+		t.Run("Should return db applied migrations for "+driver, func(t *testing.T) {
+			t.Parallel()
+			settings := makeSqlSettings(driver)
+			store := New(*settings, nil)
+
+			migrations, err := store.GetAppliedMigrations()
+			require.NoError(t, err)
+			require.NotEmpty(t, migrations)
```
Comment (agnivade) [inline store/sqlstore/store_test.go:821]: Same as above. If we can get the number somehow, we can verify the number of items in the slice.

**Review by agnivade — COMMENTED** [review]
Nice work! The UI looks amazing. Killer CSS skills :)

File: store/sqlstore/store.go [hunk 4]
```diff
@@ -353,9 +353,8 @@ func (ss *SqlStore) DriverName() string {
 	return *ss.settings.DriverName
 }
 
-func (ss *SqlStore) GetCurrentSchemaVersion() string {
-	var version string
-	_ = ss.GetMasterX().Get(&version, "SELECT Value FROM Systems WHERE Name='Version'")
+func (ss *SqlStore) getCurrentSchemaVersion() string {
```
[the remark this answers is not in the document]
Comment (isacikgoz) [reply store/sqlstore/store.go:356]: Yes, this is only used in `upgrade.go` and we are going to remove it eventually.

File: app/config.go (hunk 1 above)
Comment (isacikgoz) [reply app/config.go:363]: It was being set by the `model.CurrentVersion` constant.

File: store/sqlstore/store_test.go (hunk 2 above)
Comment (isacikgoz) [reply store/sqlstore/store_test.go:802]: Yeah, it's doable. Same for below 👍

File: app/config.go (hunk 1 above)
Comment (agnivade) [reply app/config.go:363]: Can we remove that line then?

**Review by agnivade — COMMENTED** [review]
Looks good! Just one comment https://github.com/mattermost/mattermost-server/pull/19586#discussion_r813532293

**Review by agnivade — APPROVED** [review]
Neat! Looking forward to this

**Pull request merged**